src/app/server/Dnssd.h

```diff
@@ -17,6 +17,10 @@
 
 #pragma once
 
+#include <app/server/Server.h>
```

Comment:
Not convinced this is better if you need to include `server.h`  anyway

Reply:
Only thing i'm not sure of tbh... but only moved the include from Dnssd.cpp to Dnssd.h.
I can revert this to the Server::GetICDManager if needed

Reply:
I don't mind too much. But it does add two functions a pointer and 1 additional init call in server.cpp vs a longer line of code to get the instance from the server directly.

We aren't saving a dependency or include so there isn't much gain in my opinion.

Reply:
@jmartinez-silabs once https://github.com/project-chip/connectedhomeip/pull/30661 is merged, we won't need server and can include the `ICDConfig` directly. 

with this in mind, do you still prefer the `GetICDManager` from server option?

Reply:
I guess. What was server used for that got replaced by ICDConfig?

You need icdManager for this
`SupportsFeature(Clusters::IcdManagement::Feature::kLongIdleTimeSupport)`

Reply:
`Server.h` was not explictly needed anymore with the proposed structure. I was using it to get the `AppConfg.h` to avoid including it directly here.

Since that is being replaced with `ICDConfig.h`, we can limit the include scope to `ICDConfig.h` and `ICDManager.h which would allow us to avoid using the Global getter.

Question is does this version compared to the initial one provide a benefit?

Reply:
Removing the include/dependency requirement on server is worth it then

Reply:
done